### PR TITLE
GLAPPO

### LINKED ISSUE #298: Seamless support for multiple backends
Support multiple backends (NumPy, PyTorch, JAX, CuPy, ...) by decoupling the implementation of DMD variant from the provider of linear algebra operations. This would enable running on distributed architectures, GPUs, and maybe even differentiation.

This could be achieved by inferring the backend from the type of vectors/matrices provided by the user, and writing some mapping class for functions and methods.

### Example (high level idea)
```python
from pydmd.la_mappings import svd

X = np.ndarray
svd(X)
```

```python
def svd(X):
   if isinstance(X, np.ndarray):
      return np.linalg.svd
   ...
```

Fixes #297

### PR DESCRIPTION
**Pull request by fandreuz**
This PR showcases a proof of concept for generic linear algebra in PyDMD (`cdmd.py` has been adapted to the new framework).
At the moment the supported backends are PyTorch and NumPy, but I have some work-in-progress for SciPy's sparse matrices. 

As part of this PR I'd like to discuss also the design, usability and extensibility of the `linalg` module.

Fixes #298 and #297

### CODE REVIEW

**Comment by fAndreuzzi** [comment]
TODO: replace `@` with `dot`

**Comment by fAndreuzzi** [comment]
Reverted `pseudo_hankel_matrix` to the first NumPy implementation in #270 since `sliding_window_view` is not available in PyTorch: https://github.com/mathLab/PyDMD/pull/299/files#diff-aac419e0f55b2793ecc4a3dad579a318b5cd659659882046ce71b8784022a159L246

**Comment by fAndreuzzi** [comment]
TODO:
- [ ] CDMD
- [ ] DMD Modes tuner
- [ ] DMD
- [ ] DMDBase
- [ ] DMDC
- [ ] DMDOperator
- [ ] FbDMD
- [ ] HankelDMD
- [ ] HavokDMD
- [ ] HODMD
- [ ] MRDMD
- [ ] OptDMD
- [ ] ParametricDMD
- [ ] RDMD
- [ ] SPDMD
- [ ] SubspaceDMD

**Comment by fAndreuzzi** [comment]
Twin repository: https://github.com/fAndreuzzi/glappo

File: pydmd/dmdbase.py [hunk 1]
```diff
@@ -5,11 +5,12 @@
 
 import pickle
 from copy import copy, deepcopy
-from os.path import splitext
 
 import numpy as np
+from past.utils import old_div
```
[the remark this answers is not in the document]
Comment (fandreuz) [reply pydmd/dmdbase.py]: Residual of some merge conflicts, thanks! Fixed in 6a53e49cfc406f65ac308c783d8295bb9409643a

**Pull request closed without merging**